### ISSUE TITLE
Linenumber access when calling Go function

### ISSUE THREAD

**Issue by emicklei** [robertkrimen/otto#81]
I am using the "Set" function a lot to make a DSL , e.g. otto.Set("*schedule*", Schedule).

When the programmer using my DSL passes a wrong parameter value to the *schedule* function in Javascript, I would like to report that using line/position information from the source that is run.

It does not seem to be available (or exported) by the vm (Otto). Would this be possible to have?

**Comment by emicklei** [comment]
With this extension, it is possible to get this information
```
func (self FunctionCall) Location() string {
	return self.runtime.scope.frame.location()
}
```
I think it is safe to assume that there is a scope and frame available to report the location when the custom function (in the example RotateSteeringWheel) is called.
Shall I create a PR for this?

**Comment by emicklei** [comment]
I have discovered that this location returns `unknown` (i.e file ref nil) when the function was called from Go. 

I have created a simplified example from my use case that demonstrates this (my project uses a time scheduler that calls Javascript defined functions).

```
package main

import (
	"fmt"
	"log"

	"github.com/emicklei/otto"  // this fork includes my extension
)

type CallLater struct{}

func (c CallLater) CallViaGo(call otto.FunctionCall) otto.Value {
	call.Argument(0).Call(otto.NullValue())
	return otto.NullValue()
}

func (c CallLater) Where(call otto.FunctionCall) otto.Value {
	log.Println(call.CallerLocation())
	return otto.NullValue()
}

func main() {
	o := otto.New()
	o.Set("c", new(CallLater))
	script, err := o.Compile("test.go", `
	
	function later() {
		c.Where()
	}		
	later()
	c.CallViaGo(later);
`)
	if err != nil {
		panic(err.Error())
	}
	o.Run(script)
}
```
which results in 
```
2015/05/01 19:33:43 later (test.go:4:3)
2015/05/01 19:33:43 <unknown>
```

I have not yet found a different implementation (or change of the existing otto pkg) to fix this.

**Comment by emicklei** [comment]
i spent another hour finding a solution to this (fiddling with passing frames, etc). Anybody has a suggestion how I can make this work?

**Comment by deoxxa** [comment]
I'm going to close this, since I merged your PR a few days ago. Thanks again for implementing it!

**Comment by zupa-hu** [comment]
For anyone looking, this was fixed in #115.
